packages/near-api-js/src/contract.ts

```diff
@@ -13,6 +16,52 @@ function nameFunction(name: string, body: (args?: any[]) => any) {
     }[name];
 }
 
+function validateArguments(args: object, abiFunction: AbiFunction, ajv: Ajv, abiRoot: AbiRoot) {
+    if (!isObject(args)) return;
+
+    if (abiFunction.params?.serialization_type !== AbiSerializationType.Json) {
+        throw new UnsupportedSerializationError(abiFunction.name, abiFunction.params.serialization_type);
+    }
+
+    if (abiFunction.result?.serialization_type !== AbiSerializationType.Json) {
+        throw new UnsupportedSerializationError(abiFunction.name, abiFunction.params.serialization_type);
```

Comment:
Should this match the predicate?

```suggestion
        throw new UnsupportedSerializationError(abiFunction.name, abiFunction.result?.serialization_type);
```

Reply:
Good catch!